challenges/2021-06-08-simple-2d-iterator/solutions/java/codingis4noobs2/Codingis4noobs2.java

```diff
@@ -0,0 +1,47 @@
+// 2D Array Itreator
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+class NestedIterator<T> implements Iterator<T> {
+    private T[][] array;
+    private int row;
+    private int col;
+
+    public NestedIterator(T[][] array) {
+        this.array = array;
+        row = 0;
+        col = 0;
+    }
+
+    public boolean hasNext() {
+        return row < array.length && col < array[row].length;
```

Comment:
I wonder if you actually need to check the column here. Your `next()` function seems like it would prevent `col` from ever being >= the number of columns in your input array.